File: golio/infrastructure/repository/conv/article.go
package conv

import (
	"github.com/sunjin110/folio/golio/domain/model"
	"github.com/sunjin110/folio/golio/infrastructure/repository/dto/postgres_dto"
)

func NewArticle(summary *postgres_dto.ArticleSummary, body *postgres_dto.ArticleBody, tagMap map[string]*model.ArticleTag) *model.Article {
	tags := []*model.ArticleTag{}
	for _, tagID := range summary.TagIDs {
		tag, ok := tagMap[tagID]
		if !ok {
			continue
		}
		tags = append(tags, tag)
	}
	return &model.Article{
		ID:        summary.ID,
		Title:     summary.Title,
		Body:      body.Body,
		Writer:    "",
		Tags:      tags,
		CreatedAt: summary.CreatedAt,
		UpdatedAt: summary.UpdatedAt,
	}
}

File: reolio/src/domain/model/article.ts
export interface Article {
  id: string;
  title: string;
  body: string;
  tags: ArticleTag[];
  created_at: string;
}

export class ArticleSummary {
  id!: string;
  title!: string;
  tags!: ArticleTag[];
  created_at!: string;
  updated_at!: string;
}

export class ArticleTag {
  id!: string;
  name!: string;
}
